StockTrader/src/main/java/com/restResource/StockTrader/controller/BuyTriggerController.java

```diff
@@ -0,0 +1,169 @@
+                                    .errorMessage("Insufficient Funds for the Transaction - requesting to remove: " + cost)
+                                    .build());
+                    return HttpStatus.BAD_REQUEST; //insufficient funds for the transaction.
+                }
+            }
+            stockBuyTriggerStatus.get().setStock_amount(stockAmount + stockBuyTriggerStatus.get().getStock_amount());
+            buyTriggerRepository.save(stockBuyTriggerStatus.get());
+        } else {
+            BuyTrigger buyTrigger = BuyTrigger.builder()
+                .user_id(userId)
+                .stock_amount(stockAmount)
+                .stock_symbol(stockSymbol)
+                .timestamp(LocalDateTime.now())
+                .build();
+            buyTriggerRepository.save(buyTrigger);
+        }
+
+
+
+        return HttpStatus.OK;
+    }
+
+    @PostMapping(path = "/trigger")
+    public @ResponseBody
+    HttpStatus createTriggerCost(
+            @RequestParam String userId,
+            @RequestParam String stockSymbol,
+            @RequestParam int stockCost,
+            @RequestParam int transactionNum) {
+
+        if (stockCost <= 0) {
+            throw new IllegalArgumentException(
+                    "The amount parameter must be greater than zero.");
+        }
+
+        loggingService.logUserCommand(
+                UserCommandLog.builder()
+                        .command(CommandType.SET_BUY_TRIGGER)
+                        .username(userId)
+                        .stockSymbol(stockSymbol)
+                        .transactionNum(transactionNum)
+                        .build());
+
+        buyTriggerService.start(userId, stockSymbol, stockCost, transactionNum);
+
+        return HttpStatus.ACCEPTED; //we do accepted since we cant be sure it worked but we can be sure we passed it to a thread
+    }
+
+    @PostMapping(path = "/cancel")
+    public @ResponseBody
+    HttpStatus createTriggerCost(
+            @RequestParam String userId,
+            @RequestParam String stockSymbol,
+            @RequestParam int transactionNum) {
+
+        loggingService.logUserCommand(
+                UserCommandLog.builder()
+                        .command(CommandType.CANCEL_BUY_TRIGGER)
+                        .username(userId)
+                        .stockSymbol(stockSymbol)
+                        .transactionNum(transactionNum)
+                        .build());
+
+        Optional<BuyTrigger> stockBuyTriggerStatus = buyTriggerRepository.findByUserIdAndStockSymbol(userId, stockSymbol);
+
+        if (stockBuyTriggerStatus.isPresent()) {
+            //refund the money
+            //TODO add a check here to make sure it worked
+            accountRepository.updateAccountBalance(userId, stockBuyTriggerStatus.get().getStock_amount()*stockBuyTriggerStatus.get().getStock_cost(),transactionNum, "TS1");
+            buyTriggerRepository.delete(stockBuyTriggerStatus.get());
```

Comment:
deleteById() would likely be faster

Reply:
Ok, i can change it to that, i started doing it like this because i was using a composite key and wasnt sure how to use the deleteById function work with that. I guess just pass in the composite key object with the two values?

Reply:
yup exactly